openpype/hosts/maya/api/lib.py

```diff
@@ -172,6 +174,219 @@ def maintained_selection():
+    review_instance_height = instance.data.get("review_height")
+    preset["camera"] = camera
+
+    # Tests if project resolution is set,
+    # if it is a value other than zero, that value is
+    # used, if not then the asset resolution is
+    # used
+    if review_instance_width and review_instance_height:
+        preset["width"] = review_instance_width
+        preset["height"] = review_instance_height
+    elif width_preset and height_preset:
+        preset["width"] = width_preset
+        preset["height"] = height_preset
+    elif asset_width and asset_height:
+        preset["width"] = asset_width
+        preset["height"] = asset_height
+    preset["start_frame"] = start
+    preset["end_frame"] = end
+
+    # Enforce persisting camera depth of field
+    camera_options = preset.setdefault("camera_options", {})
+    camera_options["depthOfField"] = cmds.getAttr(
+        "{0}.depthOfField".format(camera))
+
+    preset["filename"] = path
+    preset["overwrite"] = True
+
+    cmds.refresh(force=True)
+
+    refreshFrameInt = int(cmds.playbackOptions(q=True, minTime=True))
+    cmds.currentTime(refreshFrameInt - 1, edit=True)
+    cmds.currentTime(refreshFrameInt, edit=True)
+
+    # Use displayLights setting from instance
+    key = "displayLights"
+    preset["viewport_options"][key] = instance.data[key]
+
+    # Override transparency if requested.
+    transparency = instance.data.get("transparency", 0)
+    if transparency != 0:
+        preset["viewport2_options"]["transparencyAlgorithm"] = transparency
+
+    # Isolate view is requested by having objects in the set besides a
+    # camera. If there is only 1 member it'll be the camera because we
+    # validate to have 1 camera only.
+    if instance.data["isolate"] and len(instance.data["setMembers"]) > 1:
+        preset["isolate"] = instance.data["setMembers"]
+
+    # Show/Hide image planes on request.
+    image_plane = instance.data.get("imagePlane", True)
+    if "viewport_options" in preset:
+        preset["viewport_options"]["imagePlane"] = image_plane
+    else:
+        preset["viewport_options"] = {"imagePlane": image_plane}
+
+    # Disable Pan/Zoom.
+    pan_zoom = cmds.getAttr("{}.panZoomEnabled".format(preset["camera"]))
+    preset.pop("pan_zoom", None)
+    preset["camera_options"]["panZoomEnabled"] = instance.data["panZoom"]
+
+    # Need to explicitly enable some viewport changes so the viewport is
+    # refreshed ahead of playblasting.
+    keys = [
+        "useDefaultMaterial",
+        "wireframeOnShaded",
+        "xray",
+        "jointXray",
+        "backfaceCulling",
+        "textures"
+    ]
+    viewport_defaults = {}
+    for key in keys:
+        viewport_defaults[key] = cmds.modelEditor(
+            instance.data["panel"], query=True, **{key: True}
+        )
+        if preset["viewport_options"][key]:
+            cmds.modelEditor(
+                instance.data["panel"], edit=True, **{key: True}
+            )
```

Comment:
This is likely also better managed in a context manager since these are reverted afterwards - actually, doesn't `capture.py` library already offer context managers for these?